pkg/logs/tag/provider.go

```diff
@@ -28,32 +30,94 @@ var NoopProvider Provider = &noopProvider{}
 
 // provider caches a list of up-to-date tags for a given entity polling periodically the tagger.
 type provider struct {
-	entityID string
-	tags     []string
-	done     chan struct{}
-	mu       sync.Mutex
+	entityID             string
+	tags                 []string
+	done                 chan struct{}
+	forceRefresh         bool
+	waitOnFirstCall      bool
+	forceRefreshDuration time.Duration
+	taggerWarmupDuration time.Duration
+	mu                   sync.Mutex
 }
 
 // NewProvider returns a new Provider.
 func NewProvider(entityID string) Provider {
-	return &provider{
-		entityID: entityID,
-		tags:     []string{},
-		done:     make(chan struct{}),
+	p := &provider{
+		entityID:        entityID,
+		tags:            []string{},
+		done:            make(chan struct{}),
+		forceRefresh:    config.Datadog.GetBool("logs_config.k8s_wait_for_tags"),
+		waitOnFirstCall: config.Datadog.GetBool("logs_config.k8s_wait_for_tags"),
 	}
+	if p.forceRefresh {
+		p.forceRefreshDuration = config.Datadog.GetDuration("logs_config.force_tagger_call_duration") * time.Second
+		p.taggerWarmupDuration = config.Datadog.GetDuration("logs_config.tagger_warmup_duration") * time.Second
+	}
+	return p
 }
 
 // GetTags returns the list of up-to-date tags.
+// if logs_config.k8s_wait_for_tags enabled:
+//   - GetTags waits on its first call before updating the local cache for 2 seconds (default)
+//   - GetTags forces updating the local cache for 5 seconds (default)
 func (p *provider) GetTags() []string {
+	if p.getWaitOnFirstCall() {
```

Comment:
I think you could use once here: https://golang.org/pkg/sync/#Once

Reply:
thanks for the trick